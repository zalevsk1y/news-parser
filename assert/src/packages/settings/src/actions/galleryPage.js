import {types} from './index';


export const addGallery=()=>{
    return{
        type:types.gallery.TOGGLE_ADD_GALLERY,
       
    }
}
export const shortCode=(text)=>{
    return{
        type:types.gallery.CHANGE_SHORTCODE,
        text:text
    }
}
export const parameterName=(text)=>{
    return{
        type:types.gallery.CHANGE_PARAMETER_NAME,
        text:text,
    }
}

