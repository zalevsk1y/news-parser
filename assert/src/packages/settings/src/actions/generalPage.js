import  {types} from './index';

export function toggleAddSource(){
    return{
        type:types.general.TOGGLE_ADDSOURCE
    }
}

