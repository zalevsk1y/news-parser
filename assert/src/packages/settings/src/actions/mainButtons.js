import {types} from './index';
import config from "@news-parser/config"
import {oldServerData,newServerData} from '@news-parser/helpers';   

//import {store} from '../index'
function sendToServer(){
    return {
        type:types.main.SEND_REQUEST,
    }
}
export function resetToDefault(){
    return {
        type:types.main.RESET_TO_DEFAULT
    }
}
function responseFromServer(data){
    return{
        type:types.main.RECEIVE_RESPONSE,
        data:data.main
    }
}
function setAllSettings(data){
    return {
        type:types.main.SET_ALL_SETTINGS,
        settings:data.settings
    }
}

export function saveSettingsToServer (dispatch,settings,nonce){
  
    const parameters=config.emulateJSON?oldServerData('settings',settings):newServerData(settings);
    return dispatch=>{
        dispatch(sendToServer())
        return fetch(config.settingsApi.saveSettings+'&_wpnonce='+nonce,
        parameters
       )
       .then(response=>response.json())
       .then((json,error)=>{
            
           if(!error&&json){
            dispatch(responseFromServer(json));
        
           }
       })
    }
}
export function getSettingsFromServer (dispatch,nonce){
    const parameters=config.emulateJSON?oldServerData():newServerData();
    return dispatch=>{ 
        return fetch(config.settingsApi.getSettings+'&_wpnonce='+nonce,parameters)
       .then(response=>response.json())
       .then((json,error)=>{
           if(!error&&json){
            dispatch(setAllSettings(json));
           }
       })
    }
}
export function getDefaultSettingsFromServer (dispatch,nonce){
    const parameters=config.emulateJSON?oldServerData():newServerData();
    return dispatch=>{ 
        return fetch(config.settingsApi.getDefaultSettings+'&_wpnonce='+nonce,parameters)
       .then(response=>response.json())
       .then((json,error)=>{
           if(!error&&json){
            dispatch(setAllSettings(json));
           }
       })
    }
}