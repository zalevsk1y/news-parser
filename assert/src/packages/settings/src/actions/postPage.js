import  {types} from './index';


export const addThumbnail=()=>{
    return{
        type:types.post.TOGGLE_ADD_THUMBNAIL,
      
    }
}
export const parseOtherPictures=()=>{
    return{
        type:types.post.TOGGLE_PARSE_OTHER_PICTURES,
    }
}
export const showPicturesDialog=()=>{
    return{
        type:types.post.TOGGLE_SHOW_PICTURES_DIALOG,
    }
}
export const maxPictures=(number)=>{
    return{
        type:types.post.CHANGE_MAX_PICTURES,
        number:number,
    }
}

