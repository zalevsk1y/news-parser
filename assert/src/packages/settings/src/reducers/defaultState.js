
export const defaultState={ 
    main:{
        isFetching:false,
        error:false,
        message:false,
    },
    settings:{
        general:{
            addSource:false
        },
        post:{
            addThumbnail:true,
            parseOtherPictures:false,
            showPicturesDialog:true,
            maxPictures:5,
        },
        gallery:{
            addGallery:false,
            shortCode:undefined,
            parameterName:undefined
        }
}
};
