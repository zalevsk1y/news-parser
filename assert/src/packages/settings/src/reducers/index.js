import general from './generalPage.js';
import post from './postPage.js';
import gallery from './galleryPage.js';
import {main} from './main.js'
import {combineReducers} from 'redux';

const settings=combineReducers({
    general,
    post,
    gallery,
})
const  settingsReducer=combineReducers({
    main,
    settings
})

function flowLeftToRight(funcObject){
    return (state,action)=>{
        let newState=state;
        const funcArray=Object.keys(funcObject)
        for(var i=0,end=funcArray.length;i<end;i++){
            var funcName=funcArray[i];
            newState=funcObject[funcName].apply(this,[newState,action]);
        }
        return newState;
    }
}
export default settingsReducer;