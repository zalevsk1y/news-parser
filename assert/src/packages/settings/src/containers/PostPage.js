import React from 'react';
import {connect} from 'react-redux';
import {postPage} from '../actions/index';
import Switch from '../components/Switch';
import InputField from '../components/InputField';
import {POST_FIELD_NAME} from '../actions/index';
import PropTypes from 'prop-types';
import Translate from './Translate';

export const PostPage = ({settings, toggleAddThumbnail, toggleParseOtherPictures, toggleShowPicturesDialog,changeMaxPictures}) => (
    <div>
    <table className="settings-content">
        <tbody>
            <tr>
                <th>
                    <Translate>Add main image to post</Translate>
                </th>
                <td>
                    <Switch
                        className="settings-switch parse-pictures"
                        active={settings.addThumbnail}
                        onClick={toggleAddThumbnail}
                        fieldName='addThumbnail'></Switch>
                </td>
            </tr>
            <tr>
                <th>
                    <Translate>Parse all pictures</Translate> 
                </th>
                <td>

                    <Switch
                        className="settings-switch download-pictures"
                        active={settings.parseOtherPictures}
                        onClick={toggleParseOtherPictures}
                        fieldName='downloadPictures'></Switch>
                    <p className="description">
                        <Translate>Try to parse additional image from the page.More</Translate>&nbsp;
                        <a href="?page=news-parser-menu-about#news-parser-settings"><Translate>info</Translate></a>
                    </p>

                </td>
            </tr>
        </tbody>
    </table>
    <table className={"sub-settings-content "+(settings.parseOtherPictures?"visible":"hidden")}>       
        <tbody>    
        <tr>
                <th>
                    <Translate>Show select pictures dialog</Translate>
                </th>
                <td>
                    <Switch
                        className="settings-switch download-pictures"
                        active={settings.showPicturesDialog}
                        onClick={toggleShowPicturesDialog}
                        fieldName='downloadPictures'></Switch>
                    <p className="description">
                        <Translate>Show preview of all parsed images in additional window.More</Translate>&nbsp;
                        <a href="?page=news-parser-menu-about#news-parser-settings"><Translate>info</Translate></a>
                    </p>

                </td>
            </tr>
            
            <tr>
                <th>
                    <Translate>Maximum pictures to add</Translate>
                </th>
                <td>
                    <InputField
                        className="settings-class-container"
                        onEnterPress={changeMaxPictures}
                        onBlur={changeMaxPictures}
                        fieldName={'MaxPictures'}
                        value={settings.maxPictures}/>
                    <p className="description">
                        <Translate>You can specify maximum number of pictures to add to your post</Translate>
                    </p>

                </td>
            </tr>
        </tbody>
    </table>
    </div>
)


const mapStateToProps = (state, ownProps) => {
        return {
            settings: state.settings[POST_FIELD_NAME]
        }
    },
    mapDispatchToProps = (dispatch, ownProps) => {
        const toggleAddThumbnail = () => {
                dispatch(postPage.addThumbnail());
            },
            toggleParseOtherPictures = () => {
                dispatch(postPage.parseOtherPictures());
            },
            toggleShowPicturesDialog = () => {
                dispatch(postPage.showPicturesDialog());
            },
            changeMaxPictures = (number) => {
                dispatch(postPage.maxPictures(parseInt(number)))
            };
        return {toggleAddThumbnail, toggleParseOtherPictures, toggleShowPicturesDialog,changeMaxPictures}

    }

export default connect(mapStateToProps, mapDispatchToProps)(PostPage);

PostPage.propTypes={
    settings:PropTypes.shape({
        addThumbnail:PropTypes.bool,
        downloadPictures:PropTypes.bool,
        maxPictures:PropTypes.number
    }).isRequired,
    toggleAddThumbnail:PropTypes.func.isRequired,
    toggleParseOtherPictures:PropTypes.func.isRequired,
    toggleShowPicturesDialog:PropTypes.func.isRequired,
    changeMaxPictures:PropTypes.func.isRequired

}