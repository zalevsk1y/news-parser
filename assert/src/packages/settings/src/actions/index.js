import {toggleAddSource} from './generalPage';
import {addThumbnail,parseOtherPictures,showPicturesDialog,maxPictures} from './postPage';
import {addGallery,shortCode,parameterName} from './galleryPage';
import {getDefaultSettingsFromServer,saveSettingsToServer,getSettingsFromServer} from './mainButtons';

export const types={
    main:{
        SEND_REQUEST:'SEND_REQUEST',
        RECEIVE_RESPONSE:'RECEIVE_RESPONSE',
        SET_ALL_SETTINGS:'SET_ALL_SETTINGS',
        RESET_TO_DEFAULT:'RESET_TO_DEFAULT'
    },
    general:{
        TOGGLE_ADDSOURCE:'TOGGLE_ADDSOURCE'
    },
    post:{
        TOGGLE_ADD_THUMBNAIL:'TOGGLE_DOWNLOAD_PICTURES',
        TOGGLE_SHOW_PICTURES_DIALOG:'TOGGLE_SHOW_PICTURES_DIALOG',
        CHANGE_MAX_PICTURES:'CHANGE_MAX_PICTURES',
        TOGGLE_PARSE_OTHER_PICTURES:'TOGGLE_PARSE_OTHER_PICTURES'
    },
    gallery:{
        TOGGLE_ADD_GALLERY:"TOGGLE_ADD_GALLERY",
        CHANGE_SHORTCODE:"CHANGE_SHORTCODE",
        CHANGE_PARAMETER_NAME:"CHANGE_PARAMETER_NAME"
    }

}


export const GALLERY_FIELD_NAME='gallery';
export const POST_FIELD_NAME='post';
export const GENERAL_FIELD_NAME='general';

export const  generalPage={toggleAddSource};
export const  postPage={addThumbnail,parseOtherPictures,showPicturesDialog,maxPictures};
export const  galleryPage={addGallery,shortCode,parameterName};
export const  main={getDefaultSettingsFromServer,saveSettingsToServer,getSettingsFromServer}
