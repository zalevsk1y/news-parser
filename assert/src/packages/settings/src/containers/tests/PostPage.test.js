import React from 'react';
import {shallow,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {PostPage} from '../PostPage';
import {defaultState} from '../../reducers/defaultState'
import {POST_FIELD_NAME} from '../../actions'

configure({adapter:new Adapter()});
describe('Post Page',()=>{
    it('should render correctly',()=>{
        const dispatchToProps={
            toggleAddThumbnail:jest.fn(),
            toggleParseOtherPictures:jest.fn(),
            toggleShowPicturesDialog:jest.fn(),
            changeMaxPictures:jest.fn()
        }
        const component=shallow(<PostPage settings={defaultState.settings[POST_FIELD_NAME]} {...dispatchToProps} />)
        expect(component).toMatchSnapshot()
    })
})