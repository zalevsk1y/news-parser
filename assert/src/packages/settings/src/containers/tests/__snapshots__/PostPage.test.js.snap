// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Page should render correctly 1`] = `
<div>
  <table
    className="settings-content"
  >
    <tbody>
      <tr>
        <th>
          <Connect(Translate)>
            Add main image to post
          </Connect(Translate)>
        </th>
        <td>
          <Switch
            active={true}
            className="settings-switch parse-pictures"
            fieldName="addThumbnail"
            onClick={[MockFunction]}
          />
        </td>
      </tr>
      <tr>
        <th>
          <Connect(Translate)>
            Parse all pictures
          </Connect(Translate)>
        </th>
        <td>
          <Switch
            active={false}
            className="settings-switch download-pictures"
            fieldName="downloadPictures"
            onClick={[MockFunction]}
          />
          <p
            className="description"
          >
            <Connect(Translate)>
              Try to parse additional image from the page.More
            </Connect(Translate)>
             
            <a
              href="#"
            >
              <Connect(Translate)>
                info
              </Connect(Translate)>
            </a>
          </p>
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="sub-settings-content hidden"
  >
    <tbody>
      <tr>
        <th>
          <Connect(Translate)>
            Show select pictures dialog
          </Connect(Translate)>
        </th>
        <td>
          <Switch
            active={true}
            className="settings-switch download-pictures"
            fieldName="downloadPictures"
            onClick={[MockFunction]}
          />
          <p
            className="description"
          >
            <Connect(Translate)>
              Show preview of all parsed images in additional window.More
            </Connect(Translate)>
             
            <a
              href="#"
            >
              <Connect(Translate)>
                info
              </Connect(Translate)>
            </a>
          </p>
        </td>
      </tr>
      <tr>
        <th>
          <Connect(Translate)>
            Maximum pictures to add
          </Connect(Translate)>
        </th>
        <td>
          <InputField
            className="settings-class-container"
            fieldName="MaxPictures"
            onBlur={[MockFunction]}
            onEnterPress={[MockFunction]}
            value={5}
          />
          <p
            className="description"
          >
            <Connect(Translate)>
              You can specify maximum number of pictures to add to your post
            </Connect(Translate)>
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
